lsp/server/src/diagnostic/DiagnosticSettings.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+export type ProducerId = 
+    | 'misspelled-uiapi' 
+    | 'adapters-local-change-not-aware'; 
+
```

Comment:
Should `ProducerId` be generated at runtime by iterating all the producers belonging to each validator?

Reply:
removed.